src/uhi_serialization/hdf5_serialization.py

```diff
@@ -1,328 +1,415 @@
-        f[group_prefix + '/storage'].attrs['description'] = "The storage of the bins of the histogram."
+        f[group_prefix].create_group("storage")
+        f[group_prefix + "/storage"].attrs[
+            "description"
+        ] = "The storage of the bins of the histogram."
         hist_str_type = str(histogram.storage_type)
-        hist_str_type = CONSTS['storage_dict'][hist_str_type[hist_str_type.find('e') + 2 : len(hist_str_type) - 2]]
+        hist_str_type = STORAGE_MAP[
+            hist_str_type[hist_str_type.find("e") + 2 : len(hist_str_type) - 2]
+        ]
         args_dict = {}
-        args_dict['values'] = np.array(histogram.values())
+        args_dict["values"] = np.array(histogram.values())
         match hist_str_type:
-            case 'int_storage':
-                #NOTE: `int_storage` only has the stored values
+            case "int_storage":
+                # NOTE: `int_storage` only has the stored values
                 pass
-            case 'double_storage':
-                #NOTE: `double_storage` only has the stored values
+            case "double_storage":
+                # NOTE: `double_storage` only has the stored values
                 pass
-            case 'weighted_storage':
-                args_dict['variances'] = np.array(histogram.variances())
-            case 'mean_storage':
-                args_dict['variances'] = np.array(histogram.variances())
-                args_dict['counts'] = np.array(histogram.counts())
-            case 'weighted_mean_storage':
-                args_dict['variances'] = np.array(histogram.variances())
-                f1 = lambda x: x.sum_of_weights
-                f2 = lambda x: x.sum_of_weights_squared
-                args_dict['sum_of_weights'] = np.array(list(map(f1, histogram)))
-                args_dict['sum_of_weights_squared'] = np.array(list(map(f2, histogram)))
+            case "weighted_storage":
+                args_dict["variances"] = np.array(histogram.variances())
+            case "mean_storage":
+                args_dict["variances"] = np.array(histogram.variances())
+                args_dict["counts"] = np.array(histogram.counts())
+            case "weighted_mean_storage":
+                view = histogram.view()
+                assert isinstance(view, bh._internal.view.WeightedMeanView)
+                args_dict["variances"] = np.array(histogram.variances())
+                args_dict["sum_of_weights"] = view.sum_of_weights
+                args_dict["sum_of_weights_squared"] = view.sum_of_weights_squared
+
         create_storage_object(hist_str_type, f, name, args_dict)
         """
         `storage` code end
         """
     f.close()
     return f
 
+
 def read_hdf5_schema(input_file: h5py.File | Path) -> dict[str, bh.Histogram]:
-    f = input_file
-    if isinstance(input_file, Path):
-        f = h5py.File(f)
+    f = h5py.File(input_file) if isinstance(input_file, Path) else input_file
```

Comment:
This provides a statically determinable type for f (`h5py`, even though that's really just `Any` since h5py doesn't have type annotations). It's tricker in the original form (though mypy can sometimes deduce it correctly in that form too).